core/dbt/task/run.py

```diff
@@ -33,11 +37,22 @@ def run_hooks(self, adapter, hook_type, extra_context):
         # b/c psycopg2 automatically begins a transaction when a connection
         # is created.
         adapter.clear_transaction()
+        if not ordered_hooks:
+            return
 
-        for i, hook in enumerate(ordered_hooks):
+        plural = 'hook' if len(ordered_hooks) == 1 else 'hooks'
+        dbt.ui.printer.print_timestamped_line("")
+        dbt.ui.printer.print_timestamped_line(
+            'Running {} {} {}'.format(len(ordered_hooks), hook_type, plural)
+        )
+
+        for idx, hook in enumerate(ordered_hooks, start=1):
             compiled = compile_node(adapter, self.config, hook,
                                     self.manifest, extra_context)
             statement = compiled.wrapped_sql
+            dbt.ui.printer.print_timestamped_line(
+                '{} of {} START {}'.format(idx, len(ordered_hooks), statement)
```

Comment:
Can we format this like all of the other timestamped lines that dbt prints?
```
07:58:48 | 1 of 2 START hook: grant usage on schema......... [RUN]
07:58:48 | 1 of 2 OK hook: grant usage on schema......... [GRANT in 0.4s]
07:58:48 | 2 of 2 START hook: grant select on ......... [RUN]
07:58:48 | 2 of 2 OK hook: grant select on......... [GRANT in 2.8s]
```

I think we'll also want to reshape/truncate the hook SQL. It's not uncommon to do operational things (like create UDFs, or insert into an audit table in these hooks, and they currently take over the stdout!

I imagine we can do something like (pseudocode):
```
hook_text = hook.replace("\n", " ")[0:60] + "..."
```

This doesn't need to be the exact same SQL that dbt is actually running, it just needs to be a stand-in for a name! Maybe some day it will make sense to name hooks, but today is not that day :)

Let's also include a line for when the hooks completes, and a status if it's easy to grab. I think a lot of folks don't actually have insight into how long their hooks currently take, and I be it will be helpful to expose this info to them!

Here's an example hook for testing:
```
on-run-start:
 - "{{ load_udfs() }}"
```

```
-- macros/udfs.sql (imagine this did something useful....)

CREATE OR REPLACE FUNCTION no_op()
RETURNS void AS $$
$$ LANGUAGE SQL;
```

Reply:
We can also be pretty aggressive with the truncation here -- the point is just to show that hooks are running, and to indicate how long they have taken. If folks want to dig further into that, they can find the full queries in the logs!